## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 415209`
Commit: `950b4fba90213b4bdd22da4c2dd40847c8390884`

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>xenia-canary</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>xenia-canary</summary>
<pre>FAILED: CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream.c.o 
/nix/store/zcxp3wrklyg5580h40qwk4r0cbm130b8-clang-wrapper-18.1.8/bin/clang -DHAVE_AV_CONFIG_H -DNDEBUG -DUNICODE -D_LIB -D_NO_DEBUG_HEAP=1 -D_UNICODE -D_USE_MATH_DEFINES -I/build/source -I/build/source/src -I/build/source/third_party -I/build/source/third_party/FFmpeg -I/build/source/third_party/FFmpeg/compat/atomics/gcc -I/nix/store/3aw46zjyv8azwxpnpri4w6z133yv7jfx-gtk+3-3.24.49-dev/include/gtk-3.0 -I/nix/store/zba9gq8v639y0q4aixncnfpzz2zcr6va-at-spi2-core-2.56.2-dev/include/atk-1.0 -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include/glib-2.0 -I/nix/store/sjn5dv5h5qn8jzk1r0ndd9jz189x4nz0-glib-2.84.1/lib/glib-2.0/include -I/nix/store/mv0rs47h68l1bmwlrkgsyrahr1b65z2c-cairo-1.18.2-dev/include/cairo -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include/freetype2 -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include -I/nix/store/nqh6a1ar73var2f18iljnvzyszsjrf7c-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/76sh3x221l0ihlqv4gxf41i6sbjnb4fm-pango-1.56.3-dev/include/pango-1.0 -I/nix/store/64mp2lj92xxd329rijkzmxvyb19lndcb-harfbuzz-10.2.0-dev/include/harfbuzz -m64 -O3 -mavx2 -w -MD -MT CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream.c.o -MF CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream.c.o.d -o CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream.c.o -c /build/source/third_party/FFmpeg/libavcodec/bitstream.c
clang: error: unsupported option '-mavx2' for target 'aarch64-unknown-linux-gnu'

[126/653] Building C object CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avfft.c.o
FAILED: CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avfft.c.o 
/nix/store/zcxp3wrklyg5580h40qwk4r0cbm130b8-clang-wrapper-18.1.8/bin/clang -DHAVE_AV_CONFIG_H -DNDEBUG -DUNICODE -D_LIB -D_NO_DEBUG_HEAP=1 -D_UNICODE -D_USE_MATH_DEFINES -I/build/source -I/build/source/src -I/build/source/third_party -I/build/source/third_party/FFmpeg -I/build/source/third_party/FFmpeg/compat/atomics/gcc -I/nix/store/3aw46zjyv8azwxpnpri4w6z133yv7jfx-gtk+3-3.24.49-dev/include/gtk-3.0 -I/nix/store/zba9gq8v639y0q4aixncnfpzz2zcr6va-at-spi2-core-2.56.2-dev/include/atk-1.0 -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include/glib-2.0 -I/nix/store/sjn5dv5h5qn8jzk1r0ndd9jz189x4nz0-glib-2.84.1/lib/glib-2.0/include -I/nix/store/mv0rs47h68l1bmwlrkgsyrahr1b65z2c-cairo-1.18.2-dev/include/cairo -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include/freetype2 -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include -I/nix/store/nqh6a1ar73var2f18iljnvzyszsjrf7c-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/76sh3x221l0ihlqv4gxf41i6sbjnb4fm-pango-1.56.3-dev/include/pango-1.0 -I/nix/store/64mp2lj92xxd329rijkzmxvyb19lndcb-harfbuzz-10.2.0-dev/include/harfbuzz -m64 -O3 -mavx2 -w -MD -MT CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avfft.c.o -MF CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avfft.c.o.d -o CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avfft.c.o -c /build/source/third_party/FFmpeg/libavcodec/avfft.c
clang: error: unsupported option '-mavx2' for target 'aarch64-unknown-linux-gnu'

[127/653] Building C object CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avdct.c.o
FAILED: CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avdct.c.o 
/nix/store/zcxp3wrklyg5580h40qwk4r0cbm130b8-clang-wrapper-18.1.8/bin/clang -DHAVE_AV_CONFIG_H -DNDEBUG -DUNICODE -D_LIB -D_NO_DEBUG_HEAP=1 -D_UNICODE -D_USE_MATH_DEFINES -I/build/source -I/build/source/src -I/build/source/third_party -I/build/source/third_party/FFmpeg -I/build/source/third_party/FFmpeg/compat/atomics/gcc -I/nix/store/3aw46zjyv8azwxpnpri4w6z133yv7jfx-gtk+3-3.24.49-dev/include/gtk-3.0 -I/nix/store/zba9gq8v639y0q4aixncnfpzz2zcr6va-at-spi2-core-2.56.2-dev/include/atk-1.0 -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include/glib-2.0 -I/nix/store/sjn5dv5h5qn8jzk1r0ndd9jz189x4nz0-glib-2.84.1/lib/glib-2.0/include -I/nix/store/mv0rs47h68l1bmwlrkgsyrahr1b65z2c-cairo-1.18.2-dev/include/cairo -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include/freetype2 -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include -I/nix/store/nqh6a1ar73var2f18iljnvzyszsjrf7c-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/76sh3x221l0ihlqv4gxf41i6sbjnb4fm-pango-1.56.3-dev/include/pango-1.0 -I/nix/store/64mp2lj92xxd329rijkzmxvyb19lndcb-harfbuzz-10.2.0-dev/include/harfbuzz -m64 -O3 -mavx2 -w -MD -MT CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avdct.c.o -MF CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avdct.c.o.d -o CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/avdct.c.o -c /build/source/third_party/FFmpeg/libavcodec/avdct.c
clang: error: unsupported option '-mavx2' for target 'aarch64-unknown-linux-gnu'

[128/653] Building C object CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c.o
FAILED: CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c.o 
/nix/store/zcxp3wrklyg5580h40qwk4r0cbm130b8-clang-wrapper-18.1.8/bin/clang -DHAVE_AV_CONFIG_H -DNDEBUG -DUNICODE -D_LIB -D_NO_DEBUG_HEAP=1 -D_UNICODE -D_USE_MATH_DEFINES -I/build/source -I/build/source/src -I/build/source/third_party -I/build/source/third_party/FFmpeg -I/build/source/third_party/FFmpeg/compat/atomics/gcc -I/nix/store/3aw46zjyv8azwxpnpri4w6z133yv7jfx-gtk+3-3.24.49-dev/include/gtk-3.0 -I/nix/store/zba9gq8v639y0q4aixncnfpzz2zcr6va-at-spi2-core-2.56.2-dev/include/atk-1.0 -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include -I/nix/store/wgnrq9jcvlnsz309a8hf1zsngvvqm9k1-glib-2.84.1-dev/include/glib-2.0 -I/nix/store/sjn5dv5h5qn8jzk1r0ndd9jz189x4nz0-glib-2.84.1/lib/glib-2.0/include -I/nix/store/mv0rs47h68l1bmwlrkgsyrahr1b65z2c-cairo-1.18.2-dev/include/cairo -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include/freetype2 -I/nix/store/6q2irlvdzypi52k20yxl4n7rpgsn47bh-freetype-2.13.3-dev/include -I/nix/store/nqh6a1ar73var2f18iljnvzyszsjrf7c-gdk-pixbuf-2.42.12-dev/include/gdk-pixbuf-2.0 -I/nix/store/76sh3x221l0ihlqv4gxf41i6sbjnb4fm-pango-1.56.3-dev/include/pango-1.0 -I/nix/store/64mp2lj92xxd329rijkzmxvyb19lndcb-harfbuzz-10.2.0-dev/include/harfbuzz -m64 -O3 -mavx2 -w -MD -MT CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c.o -MF CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c.o.d -o CMakeFiles/libavcodec.dir/build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c.o -c /build/source/third_party/FFmpeg/libavcodec/bitstream_filter.c
clang: error: unsupported option '-mavx2' for target 'aarch64-unknown-linux-gnu'
ninja: build stopped: subcommand failed.
ERROR: ninja failed with one or more errors.</pre>
</details>
</details>

--- a/pkgs/by-name/xe/xenia-canary/package.nix
+++ b/pkgs/by-name/xe/xenia-canary/package.nix
@@ -1,0 +1,102 @@
+{
+  lib,
+  python3,
+  gtk3,
+  lz4,
+  SDL2,
+  pkg-config,
+  vulkan-loader,
+  ninja,
+  cmake,
+  libuuid,
+  wrapGAppsHook3,
+  makeDesktopItem,
+  copyDesktopItems,
+  llvmPackages_18,
+  autoPatchelfHook,
+  unstableGitUpdater,
+  fetchFromGitHub,
+}:
+llvmPackages_18.stdenv.mkDerivation {
+  pname = "xenia-canary";
+  version = "0-unstable-2025-06-07";
+
+  src = fetchFromGitHub {
+    owner = "xenia-canary";
+    repo = "xenia-canary";
+    fetchSubmodules = true;
+    rev = "422517c673bba086c2b857946ae5a37ee35b8e50";
+    hash = "sha256-88GHKXURfN8vaVNN7wKn562b6FvsIm/sTcUgtuhvVxM=";
+  };
+
+  dontConfigure = true;
+
+  nativeBuildInputs = [
+    python3
+    pkg-config
+    ninja
+    cmake
+    wrapGAppsHook3
+    copyDesktopItems
+    autoPatchelfHook
+    libuuid
+  ];
+
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error=unused-result"
+  ];
+
+  buildInputs = [
+    gtk3
+    lz4
+    SDL2
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    python3 xenia-build setup
+    python3 xenia-build build --config=release -j $NIX_BUILD_CORES
+    runHook postBuild
+  '';
+
+  runtimeDependencies = [
+    vulkan-loader
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "xenia_canary";
+      desktopName = "Xenia Canary";
+      genericName = "Xbox 360 Emulator";
+      exec = "xenia_canary";
+      comment = "Xbox 360 Emulator Research Project";
+      icon = "xenia-canary";
+      startupWMClass = "Xenia_canary";
+      categories = [
+        "Game"
+        "Emulator"
+      ];
+      keywords = [ "xbox" ];
+    })
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    find ./build/bin -mindepth 3 -maxdepth 3 -type f -executable -exec install -Dm755 {} -t $out/bin \;
+    for width in 16 32 48 64 128 256; do
+      install -Dm644 assets/icon/$width.png $out/share/icons/hicolor/''${width}x''${width}/apps/xenia-canary.png
+    done
+    runHook postInstall
+  '';
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "Xbox 360 Emulator Research Project";
+    homepage = "https://github.com/xenia-canary/xenia-canary";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ tuxy ];
+    mainProgram = "xenia_canary";
+    platforms = [ "x86_64-linux" ];
+  };
+}

